### PR TITLE
DEV-312-Update-LM-contracts

### LINKED ISSUE #41: Remove `claim` function from the claimFacet
Hi, we had briefly discussed that we will be airdropping earned vape rewards, but while doing the subgraph, I found out that we are still using the old claim function. Shall we remove it?

### PR DESCRIPTION
**Pull request by royvardhan**
Fixes #41 

1. Updated the claim facet and the diamond initializer. 
2. Removed claim facet tests that were connected to the `claim()` function and were failing since the function has been removed. 
3. Also updated the generate full abi script that does not depend on deployments anymore.
4. Ordered events according to their relevance in the facet, as discussed in the meeting.

### CODE REVIEW

File: src/facets/BoostFacet.sol [hunk 1]
```diff
@@ -18,13 +18,15 @@ error BoostFacet__InvalidFeeReceivers();
 /// @notice Facet in charge of point's boosts
 /// @dev Utilizes 'LDiamond', 'AppStorage'
 contract BoostFacet {
+
+    /// @notice Ordering of the events are according to their relevance in the facet
     event ClaimBoost(
-        address indexed _user,
         uint256 _seasonId,
```
Comment (parth-15) [inline src/facets/BoostFacet.sol:24]: Should we prefer to use `indexed` for seasonId?

File: src/facets/MiningPassFacet.sol [hunk 2]
```diff
@@ -18,7 +18,8 @@ error MiningPassFacet__SeasonEnded();
 contract MiningPassFacet {
     AppStorage s;
 
-    event MiningPassPurchase(address indexed user, uint256 indexed tier, uint256 fee);
+    /// @notice Ordering of the events are according to their relevance in the facet
+    event MiningPassPurchase(uint256 seasonId, address indexed user, uint256 indexed tier, uint256 fee);
```
Comment (parth-15) [inline src/facets/MiningPassFacet.sol:22]: same here? should `seasonid` be indexed?

File: src/facets/BoostFacet.sol (hunk 1 above)
Comment (royvardhan) [reply src/facets/BoostFacet.sol:24]: We can still catch them inside subgraph and update entities.. @mejiasd3v  @Thehitesh172 thoughts?

Comment (mejiasd3v) [reply src/facets/BoostFacet.sol:24]: yes, we should always index user and seasonId

File: src/facets/BoostFacet.sol [hunk 3]
```diff
@@ -18,13 +18,15 @@ error BoostFacet__InvalidFeeReceivers();
...
+        address indexed _user,
+        uint256 boostLevel,
```
Comment (mejiasd3v) [inline src/facets/BoostFacet.sol:26]: please remove "_" from this topics definitions

File: src/facets/DepositFacet.sol [hunk 4]
```diff
@@ -19,6 +19,8 @@ error DepositFacet__InvalidMiningPass();
 /// @notice Facet in charge of depositing VPND tokens
 /// @dev Utilizes 'LDiamond', 'AppStorage' and 'LPercentages'
 contract DepositFacet {
+    
+    /// @notice Ordering of the events are according to their relevance in the facet
     event Deposit(uint256 indexed seasonId, address indexed user, uint256 amount);
```
Comment (mejiasd3v) [inline src/facets/DepositFacet.sol:24]: please index user

File: src/facets/UnlockFacet.sol [hunk 5]
```diff
@@ -14,7 +14,8 @@ error UnlockFacet__InvalidFeeReceivers();
 error UnlockFacet__InvalidUnlock();
 
 contract UnlockFacet {
-    event Unlocked(address indexed user, uint256 amount, uint256 seasonId, uint256 unlockFee);
+    /// @notice Ordering of the events are according to their relevance in the facet
+    event Unlocked(uint256 seasonId, address indexed user, uint256 amount, uint256 unlockFee);
```
Comment (mejiasd3v) [inline src/facets/UnlockFacet.sol:18]: please index seasonId

File: src/facets/WithdrawFacet.sol [hunk 6]
```diff
@@ -18,7 +18,9 @@ contract WithdrawFacet {
     //////////////
     /// EVENTS ///
     //////////////
-    event WithdrawUnlockedVPND(uint256 amount, address indexed to, uint256 seasonId);
+
+    /// @notice Ordering of the events are according to their relevance in the facet
+    event WithdrawUnlockedVPND(uint256 seasonId, address indexed to, uint256 amount);
```
Comment (mejiasd3v) [inline src/facets/WithdrawFacet.sol:23]:
```suggestion
    event WithdrawUnlockedVPND(uint256 indexed seasonId, address indexed user, uint256 amount);
```

**Comment by royvardhan** [comment]
PR comments resolved:
Indexing seasonId and removed "_" from the event definitions

**Pull request merged**